src/internal/github/assets.go

```diff
@@ -18,51 +15,25 @@ type Platform struct {
 	Arch string
 }
 
-func DownloadAssetContents(ctx context.Context, downloadURL string) ([]byte, error) {
-	httpClient := httpInternal.GetHttpRetryClient()
-
-	log.Printf("Downloading asset, url: %s", downloadURL)
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, downloadURL, nil)
+func (c Client) DownloadAssetContents(downloadURL string) ([]byte, error) {
```

Comment:
I still think we should pass a context here so we can do context propogation and timeouts if we want to later on.

Then we could have the http client use that context too.

Reply:
Context is handled in the Client, and the client should only live as long as it needs, so im fine with this. especially when it helps with context cancellation